Result/ResultType.swift

```diff
@@ -65,9 +71,9 @@ public extension ResultType {
 }
 
 /// Protocol used to constrain `tryMap` to `Result`s with compatible `Error`s.
-public protocol ErrorTypeConvertible: ErrorType {
+public protocol ErrorTypeConvertible: ResultErrorType {
 	typealias ConvertibleType = Self
-	static func errorFromErrorType(error: ErrorType) -> ConvertibleType
+	static func errorFromErrorType(error: ResultErrorType) -> Self
```

Comment:
@ikesyo It looks like it is. The thing is that you have to explicitly put there `Self` in implementation. Try to install latest 3.0 toolchain and check. In case I'm wrong, I would be happy to remove it.